src/sql/magic.py

```diff
@@ -180,6 +181,26 @@ def _mutex_autopandas_autopolars(self, change):
                 setattr(self, other, False)
                 print(f"Disabled '{other}' since '{change['name']}' was enabled.")
 
+    def check_random_arguments(self, line="", cell=""):
+        # Split the line into tokens separated by whitespace
+        tokens = line.split()
+        arguments = []
+
+        # Iterate through the tokens to separate arguments and SQL code
```

Comment:
As @edublancas, pointed out, this logic will fail in case of some SQL comments because they start with `--`.  One approach can be, as soon as you encounter non `--` token after encountering some `--` token, you can `break` the loop.